src/utils/emissions/HelpersEnergy.cpp

```diff
@@ -130,9 +136,207 @@ HelpersEnergy::compute(const SUMOEmissionClass /* c */, const PollutantsInterfac
+    if (e != PollutantsInterface::ELEC) {
+        return 0.;
+    }
+        if (param == 0) {
+        param = &myDefaultParameter;
+    }
 
+    //Inverse formula for the function compute()
+    //Computes the achievable acceleartion using the given speed and amount of consumed electric power
+    //It does not consider loss through friction by radial force and the recuperation efficiency dependent on |a| (eta_reuperation is considered constant)
+    // Prest = const1*a + const2*a^2 + const3*a^3
+
+    const double mass = param->find(SUMO_ATTR_VEHICLEMASS)->second;
+    double const1, const2, const3;
+    double Prest;
+    int numX;
+    double x1, x2, x3;
+
+    if (P > 0) {
+        // Assumption: Efficiency of myPropulsionEfficiency when accelerating
+        Prest = P*3600 * param->find(SUMO_ATTR_PROPULSIONEFFICIENCY)->second;
+    }
+    else {
+        // Assumption: Efficiency of myRecuperationEfficiency when recuperating
+        Prest = P*3600 / param->find(SUMO_ATTR_RECUPERATIONEFFICIENCY)->second;
+    }
+
+    // calculate power drop due to a potential energy difference
+    Prest -= mass * 9.81 * sin(DEG2RAD(slope)) * (v);
+    const1 = mass * 9.81 * sin(DEG2RAD(slope)) * (TS);
+
+    // Power loss through Roll resistance [W]
+    //                    ... (fabs(veh.getSpeed())>=0.01) = 0, if vehicle isn't moving
+    // EnergyLoss,Tire = c_R [-] * F_N [N] * s [m]
+    //                    ... with c_R = ~0.012    (car tire on asphalt)
+    //                    ... with F_N [N] = myMass [kg] * g [m/s^2]
+    Prest -= param->find(SUMO_ATTR_ROLLDRAGCOEFFICIENT)->second * 9.81 * mass * v;
+    const1 += param->find(SUMO_ATTR_ROLLDRAGCOEFFICIENT)->second * 9.81 * mass * (TS);
+    
+    //Constant loads are omitted. We assume P as the max limit for the main traction drive. Constant loads are often covered by an auxiliary drive
+    //Power loss through constant loads (e.g. A/C) [W]
+    //Prest -= param->find(SUMO_ATTR_CONSTANTPOWERINTAKE)->second / TS;
+
+    // kinetic energy difference of vehicle
+    const1 += 0.5 * mass * (2 * v);
+    const2 = 0.5 * mass * (TS);
+
+    // add rotational energy diff of internal rotating elements
+    const1 += param->find(SUMO_ATTR_INTERNALMOMENTOFINERTIA)->second * (2 * v);
+    const2 += param->find(SUMO_ATTR_INTERNALMOMENTOFINERTIA)->second * (TS);
+
+    // Energy loss through Air resistance [Ws]
+    // Calculate energy losses:
+    // EnergyLoss,Air = 1/2 * rho_air [kg/m^3] * myFrontSurfaceArea [m^2] * myAirDragCoefficient [-] * v_Veh^2 [m/s] * s [m]
+    //                    ... with rho_air [kg/m^3] = 1,2041 kg/m^3 (at T = 20C)
+    //                    ... with s [m] = v_Veh [m/s] * TS [s]
+    Prest -= 0.5 * 1.2041 * param->find(SUMO_ATTR_FRONTSURFACEAREA)->second * param->find(SUMO_ATTR_AIRDRAGCOEFFICIENT)->second * (v * v * v);
```

Comment:
Precompute the common part. In general, repetitive use of `param->find(SUMO_ATTR_FRONTSURFACEAREA)->second` will be probably slower than using a precomputed variable (but it might be optimised away, who knows).